bundles/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/internal/handler/SerialOceanicThingHandler.java

```diff
@@ -17,45 +17,48 @@
 import java.io.OutputStream;
 import java.text.SimpleDateFormat;
 import java.util.Arrays;
-import java.util.Enumeration;
+import java.util.TooManyListenersException;
+import java.util.stream.Collectors;
 
 import org.apache.commons.lang3.StringUtils;
+import org.eclipse.jdt.annotation.NonNull;
 import org.openhab.binding.oceanic.internal.SerialOceanicBindingConfiguration;
 import org.openhab.binding.oceanic.internal.Throttler;
+import org.openhab.core.io.transport.serial.PortInUseException;
+import org.openhab.core.io.transport.serial.SerialPort;
+import org.openhab.core.io.transport.serial.SerialPortEvent;
+import org.openhab.core.io.transport.serial.SerialPortEventListener;
+import org.openhab.core.io.transport.serial.SerialPortIdentifier;
+import org.openhab.core.io.transport.serial.SerialPortManager;
+import org.openhab.core.io.transport.serial.UnsupportedCommOperationException;
 import org.openhab.core.thing.Thing;
 import org.openhab.core.thing.ThingStatus;
 import org.openhab.core.thing.ThingStatusDetail;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import gnu.io.CommPortIdentifier;
-import gnu.io.NoSuchPortException;
-import gnu.io.PortInUseException;
-import gnu.io.RXTXCommDriver;
-import gnu.io.SerialPort;
-import gnu.io.UnsupportedCommOperationException;
-
 /**
  * The {@link SerialOceanicThingHandler} implements {@link OceanicThingHandler} for an Oceanic water softener that is
  * directly connected to a serial port of the openHAB host
  *
  * @author Karel Goderis - Initial contribution
  */
-public class SerialOceanicThingHandler extends OceanicThingHandler {
+public class SerialOceanicThingHandler extends OceanicThingHandler implements SerialPortEventListener {
 
     private static final long REQUEST_TIMEOUT = 10000;
     private static final int BAUD = 19200;
 
     private final Logger logger = LoggerFactory.getLogger(SerialOceanicThingHandler.class);
 
+    private final SerialPortManager serialPortManager;
     private SerialPort serialPort;
-    private CommPortIdentifier portId;
     private InputStream inputStream;
     private OutputStream outputStream;
     private SerialPortReader readerThread;
 
-    public SerialOceanicThingHandler(Thing thing) {
+    public SerialOceanicThingHandler(@NonNull Thing thing, SerialPortManager serialPortManager) {
```

Comment:
The `@NonNull` annotation should not be used, see:

https://www.openhab.org/docs/developer/guidelines.html#null-annotations

```suggestion
    public SerialOceanicThingHandler(Thing thing, SerialPortManager serialPortManager) {
```